feature-pitches/002-file-based-workflows/002-file-based-workflows.md

```diff
@@ -0,0 +1,109 @@
+
+
+### Misc Rabbit Holes
+
+**How do you handle when a workflow step generates many files?**
+
+As is the state of the current workflow functionality, inputs and outputs may be directories, rather than files, to support multiple files. However we don’t support any kind of wildcard syntax.
+
+**How does this feature mesh with the current workflow parameter file functionality?**
+
+Currently, the user has the option of creating a file specifying workflow parameters, and passing this in to `renku workflow execute` to override the default parameters. In the future, the combination of this parameter file and the workflow definition file could develop into a sort of base workflow + several “variation” files defining a set of related workflow experiments. But work into this direction is _out of scope_ of this pitch. 
+
+
+**Should one workflow file be able to reference another workflow file? (i.e. to compose workflows between files)**
+
+My default is *no*, keep it simple for the scope of this pitch.
+
+**Potential for Un-executable `Plans`**
+
+In the current `renku run` implementation, a `plan` is only created after the command executed successfully. By importing and creating the `plan` upon executing the workflow file, the user could potentially end up with a non-executable `plan` in their project. This sounds non-ideal, however, it doesn’t really have any implications for the user, so it’s up to the team for how they would like to handle this behavior.
+
+## 🙅‍♀️ No-gos
+
+There are a lot of feature-rich workflow tools out there already. We’re not trying to make the *best* workflow tool, and not even a comprehensive one. If we can get a researcher started using workflows for the first time because simple workflows are easy to create in Renku, and then that user switches to a different workflow tool that suits them better, that’s a win for Open Research!
+
+We’re not trying to make a comprehensive workflow tool with lots of functionality. We’re trying to make the *minimal* workflow tool a user can learn, so they can take their research project to the next level of reproducibility. So, the file-based workflow syntax need only support a minimal set of workflow functionality.
+
+Specifically, expressing `renku workflow iterate` in the workflow file definition syntax is out of scope, along with any kinds of looping, branching, and wildcard syntax.
```

Comment:
The idea of `iterate` is anyway to take an existing workflow definition and iterate through parameters defined therein - so I see it fully compatible with the feature laid out in this pitch by default.

Reply:
From my discussion with Ralf, my understanding is that this is not trivial to implement - @Panaetius ?